packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -371,6 +344,17 @@ export default function MainFeedLayout({
     }
   }, [sortingEnabled, selectedAlgo, loadedSettings, loadedAlgo]);
 
+  useEffect(() => {
+    if (
+      defaultFeed !== null &&
+      feedName !== null &&
+      feedName !== defaultFeed &&
+      !getShouldRedirect(isMyFeed, !!user)
+    ) {
+      updateDefaultFeed(feedName);
+    }
+  }, [defaultFeed, feedName]);
```

Comment:
What's this?

Reply:
I think this was to set the default feed if the user has my-feed to make it automatic. Though I just moved it downwards to make all use effects be at before rendering.

Reply:
I advise to add proper explanation or refactor into its own hook with explicit name